core/dbt/tracking.py

```diff
@@ -453,7 +454,14 @@ def track_run(run_command=None):
         track_invocation_end(invocation_context, result_type="ok")
     except (NotImplementedError, FailedToConnectError) as e:
         fire_event(MainEncounteredError(exc=str(e)))
+        if not isinstance(e, dbtException):
+            fire_event(MainStackTrace(stack_trace=traceback.format_exc()))
```

Comment:
Emulating this logic:
https://github.com/dbt-labs/dbt-core/blob/4d0ee2fc475cf7bc20caef20bb7a278a2c287c13/core/dbt/main.py#L151-L155